akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRecoveringEvents.scala

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.persistence.typed.internal
+
+import akka.actor.typed.{ Behavior, PostStop, Signal }
+import akka.actor.typed.scaladsl.{ ActorContext, Behaviors, TimerScheduler }
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.persistence.JournalProtocol._
+import akka.persistence._
+import akka.persistence.typed.internal.EventsourcedBehavior.InternalProtocol._
+import akka.persistence.typed.internal.EventsourcedBehavior._
+
+import scala.concurrent.duration.FiniteDuration
+import scala.util.control.NonFatal
+
+/***
+ * INTERNAL API
+ *
+ * Third (of four) behavior of an PersistentBehavior.
+ *
+ * In this behavior we finally start replaying events, beginning from the last applied sequence number
+ * (i.e. the one up-until-which the snapshot recovery has brought us).
+ *
+ * Once recovery is completed, the actor becomes [[EventsourcedRunning]], stashed messages are flushed
+ * and control is given to the user's handlers to drive the actors behavior from there.
+ *
+ * See next behavior [[EventsourcedRunning]].
+ * See previous behavior [[EventsourcedRecoveringSnapshot]].
+ */
+@InternalApi
+private[persistence] object EventsourcedRecoveringEvents {
+
+  @InternalApi
+  private[persistence] final case class RecoveringState[State](
+    seqNr:               Long,
+    state:               State,
+    eventSeenInInterval: Boolean = false
+  )
+
+  def apply[C, E, S](
+    setup: EventsourcedSetup[C, E, S],
+    state: RecoveringState[S]
+  ): Behavior[InternalProtocol] =
+    new EventsourcedRecoveringEvents(setup).createBehavior(state)
+
+}
+
+@InternalApi
+private[persistence] class EventsourcedRecoveringEvents[C, E, S](override val setup: EventsourcedSetup[C, E, S])
```

Comment:
You keep ignoring my comments about the naming here?
Terminology:
recovery = snapshot + replay

Meaning that this should be named EventsourcedReplayingEvents (and probably several more things in this file recovering -> replaying)

Reply:
Sorry, I misunderstood your comment about it I think. I'm fine with renaming to Replaying

Reply:
Thought it was only about log statements for some reason which I changed

Reply:
Addressed now, sorry, ignoring was not intentional